airflow/utils/dag_processing.py

```diff
@@ -352,7 +352,9 @@ def start(self):
             args=(
                 self._dag_directory,
                 self._max_runs,
-                self._processor_factory,
+                # getattr allows pickling by making _processor_factory get
+                # treated as a function, not an instance method.
+                getattr(self, "_processor_factory"),
```

Comment:
LOLwhat. Thanks pickle.

Is _processor_factory marked as @staticmethod? If not it should be, and then probably this becomes

```suggestion
                type(self)._processor_factory,
```

Reply:
@ashb I agree, for purposes of consistency. But I actually think `getattr` is more clear. IMO `type()` makes it look like it's a class attribute, which isn't the case.

Reply:
Actually, that's not true. Apparently pylint expects `type(self)` to access an explicitly declared class attribute, whereas `getattr` has no such requirement. Since `_processor_factory` is declared at runtime, and is really a dynamic instance attribute, I think it makes sense to stay with `getattr`.

Also, I tried declaring `_processor_factory` at the class level and using `type(self)._processor_factory`, but it lead to an error in the child process. This didn't seem worth debugging just to use `type(self)`.